### PR TITLE
feat(devinfo): implement From<MountInfoError> for std::io::Error

### PR DESCRIPTION
**Pull request by niladrih**

### CODE REVIEW

File: devinfo/src/mountinfo/error.rs [hunk 1]
```diff
@@ -65,3 +65,9 @@ impl From<semver::Error> for MountInfoError {
         Self::Semver(semver_err)
     }
 }
+
+impl From<MountInfoError> for std::io::Error {
+    fn from(err: MountInfoError) -> Self {
+        std::io::Error::new(std::io::ErrorKind::Other, err)
```
Comment (tiagolobocastro) [inline devinfo/src/mountinfo/error.rs]: should we match on the various error types and build a more precise error?

Comment (niladrih) [reply devinfo/src/mountinfo/error.rs]: Oh, I'm sorry that was lazy of me. Sure!

File: devinfo/src/mountinfo/error.rs [hunk 2]
```diff
@@ -65,3 +65,17 @@ impl From<semver::Error> for MountInfoError {
...
     }
 }
+
+impl From<MountInfoError> for std::io::Error {
+    fn from(error: MountInfoError) -> Self {
+        use std::io::{Error, ErrorKind};
+
+        match error {
+            MountInfoError::Io(std_io_err) => std_io_err,
+            MountInfoError::InconsistentRead { .. } => Error::new(ErrorKind::Other, error),
```
Comment (tiagolobocastro) [inline devinfo/src/mountinfo/error.rs]: Maybe busy better to explain the inconsistent read? 

```suggestion
            MountInfoError::InconsistentRead { .. } => Error::new(ErrorKind::ResourceBusy, error),
```

**Review by tiagolobocastro — APPROVED** [review]

File: devinfo/src/mountinfo/error.rs (hunk 1 above)
Comment (tiagolobocastro) [reply devinfo/src/mountinfo/error.rs]: Yeah this was the whole reason why I suggest to do this, so we can build a "better" error

File: devinfo/src/mountinfo/error.rs (hunk 2 above)
Comment (niladrih) [reply devinfo/src/mountinfo/error.rs]: Hmm, yeah that actually fits better.

Comment (niladrih) [reply devinfo/src/mountinfo/error.rs]: ResourceBusy is an unstable feature at the moment unfortunately.

Comment (tiagolobocastro) [reply devinfo/src/mountinfo/error.rs]: Damn, maybe InvalidData then

Comment (tiagolobocastro) [reply devinfo/src/mountinfo/error.rs]: Or whichever one you prefer then

Comment (niladrih) [reply devinfo/src/mountinfo/error.rs]: Sure, using InvalidData. Putting in a TODO for changing it some time in the future.

**Review by abhilashshetty04 — APPROVED** [review]

**Pull request merged**